tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -198,3 +220,187 @@ def check_status(node_id, status):
+        self.logger.info(
+            f"moving partition {to_move_tp}/{to_move_p} - {details['replicas']} -> {new_replicas}"
+        )
+
+        admin.set_partition_replicas(topic=to_move_tp,
+                                     partition=to_move_p,
+                                     replicas=new_replicas)
+        # moving partition should be present in moving list
+        wait_until(lambda: self._partitions_moving(),
+                   timeout_sec=15,
+                   backoff_sec=1)
+
+        self.logger.info(f"decommissioning node: {to_decommission}", )
+        admin.decommission_broker(to_decommission)
+
+        def check_status(node_id, status):
+            brokers = admin.get_brokers()
+            for broker in brokers:
+                if broker['node_id'] == node_id:
+                    return broker['membership_status'] == status
+
+            return False
+
+        wait_until(lambda: check_status(to_decommission, 'draining'),
+                   timeout_sec=15,
+                   backoff_sec=1)
+
+        wait_until(lambda: self._partitions_moving(),
+                   timeout_sec=15,
+                   backoff_sec=1)
+
+        # recommission broker
+        admin.recommission_broker(to_decommission)
+        wait_until(lambda: check_status(to_decommission, 'active'),
+                   timeout_sec=15,
+                   backoff_sec=1)
+
+        def one_left_moving():
+            reconfigurations = admin.list_reconfigurations()
+            return len(reconfigurations) == 1
+
+        wait_until(one_left_moving, timeout_sec=15, backoff_sec=1)
+
+    @cluster(num_nodes=7, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_recommissioning_multi_nodes(self):
+        self.start_redpanda(num_nodes=5)
+        self._create_topics()
+
+        self.start_producer(1)
+        self.start_consumer(1)
+        self.await_startup()
+        admin = Admin(self.redpanda)
+
+        brokers = admin.get_brokers()
+        to_decommission_1 = random.choice(brokers)['node_id']
+        to_decommission_2 = to_decommission_1
+
+        while to_decommission_1 == to_decommission_2:
+            to_decommission_2 = random.choice(brokers)['node_id']
+
+        # throttle recovery
+        rpk = RpkTool(self.redpanda)
+        rpk.cluster_config_set("raft_learner_recovery_rate", str(1))
+
+        self.logger.info(f"decommissioning node: {to_decommission_1}", )
+        admin.decommission_broker(to_decommission_1)
+        self.logger.info(f"decommissioning node: {to_decommission_2}", )
+        admin.decommission_broker(to_decommission_2)
+
+        def check_status(node_id, status):
+            brokers = admin.get_brokers()
+            for broker in brokers:
+                if broker['node_id'] == node_id:
+                    return broker['membership_status'] == status
+
+            return False
+
+        wait_until(lambda: check_status(to_decommission_1, 'draining'),
```

Comment:
nit: looks like this is a frequent enough pattern to encapsulate in a `wait_until_status(node_id, status)` function